lib/source/file.js

```diff
@@ -0,0 +1,167 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const fs = require('fs');
+
+class FileParser {
+  constructor() {
+    this.properties = Object.create(null);
+  }
+
+  /**
+   * Parse the file contents
+   * @param {Buffer} data
+   */
+  update(data) {
+    let properties = {};
```

Comment:
Might want to use `Object.create(null)` here so results of consistent regardless of if `update` is called or not.